apps/snapodds/src/services/tracking/segment/segment-analytics.service.ts

```diff
@@ -0,0 +1,112 @@
+import { HttpClient } from '@angular/common/http';
+import { Inject, Injectable, NgZone } from '@angular/core';
+import * as UAParser from 'ua-parser-js';
+
+import { environment } from '../../../environments/environment';
+import { ApplicationConfigService } from '../../config/application-config.service';
+import { CustomerApplicationConfigService } from '../../config/customer-application-config.service';
+import { LoggerService } from '../../logger/logger.service';
+import { UA_PARSER } from '../../tokens/ua-parser-token';
+import { AnalyticsEventType } from '../analytics-event-type';
+import { AnalyticsProperties } from '../analytics-properties';
+import { AnalyticsService } from '../analytics.service';
+import { SnapOddsOpenProperties } from '../properties/snap-odds-open-properties';
+import { SnapViewSnapFailedProperties } from '../properties/snap-view-snap-failed-properties';
+import { SnapViewSnapResultProperties } from '../properties/snap-view-snap-result-properties';
+import { SnapViewSnapWithoutResultProperties } from '../properties/snap-view-snap-without-result-properties';
+import { SegmentTrackEvent } from './segment-track-event';
+
+@Injectable()
+export class SegmentAnalyticsService implements AnalyticsService {
+  readonly serviceUrl = 'https://api.segment.io/v1/track';
+
+  constructor(
+    private readonly http: HttpClient,
+    private readonly ngZone: NgZone,
+    private readonly applicationConfigService: ApplicationConfigService,
+    private readonly customerApplicationConfigService: CustomerApplicationConfigService,
+    private readonly logger: LoggerService,
+    @Inject(UA_PARSER) private readonly uaParser: UAParser
+  ) {}
+
+  private trackEvent(event: AnalyticsEventType, properties?: AnalyticsProperties): void {
+    const writeKey = this.customerApplicationConfigService.getSegmentWriteKey() ?? 'a';
```

Comment:
I think that we should do nothing is segment write key is not defined.

Reply:
done